catalog/solr/catalog-solr-offline-gazetteer/src/main/java/ddf/catalog/solr/offlinegazetteer/SyncCatalogCommand.java

```diff
@@ -0,0 +1,135 @@
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.apache.karaf.shell.api.console.Session;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.codice.ddf.security.Security;
+import org.codice.solr.client.solrj.SolrClient;
+import org.codice.solr.client.solrj.UnavailableSolrException;
+import org.codice.solr.factory.SolrClientFactory;
+import org.opengis.filter.sort.SortOrder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Service
+@Command(
+  scope = "offline-solr-gazetteer",
+  name = "synccatalog",
+  description = "Syncs all catalog items to the offline solr gazetteer core"
+)
+public class SyncCatalogCommand implements Action {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(RemoveAllOfflineSolrGazetteerCommand.class);
+  public static final int PARTITION_SIZE = 256;
+
+  @Reference private SolrClientFactory clientFactory;
+
+  @Reference private Session session;
+
+  @Reference private CatalogFramework catalogFramework;
+  @Reference private FilterBuilder filterBuilder;
+  @Reference private Security security;
+
+  private final RetryPolicy retryPolicy =
+      new RetryPolicy()
+          .retryOn(ImmutableList.of(UnavailableSolrException.class, SolrServerException.class))
+          .withMaxDuration(5, TimeUnit.SECONDS)
+          .withBackoff(50, 1_000, TimeUnit.MILLISECONDS);
+
+  @Override
+  public Object execute() throws Exception {
+    return security.runWithSubjectOrElevate(this::executeWithSubject);
+  }
+
+  public Object executeWithSubject() throws Exception {
+    SolrClient solrClient =
+        clientFactory.newClient(OfflineGazetteerPlugin.STANDALONE_GAZETTEER_CORE_NAME);
+
+    Failsafe.with(retryPolicy).get(() -> solrClient.ping());
+
+    Iterable<Result> iterable =
+        ResultIterable.resultIterable(catalogFramework, getGazetteerFilter());
+
+    session.getConsole().println("Starting sync...");
+    long counter = 0;
+    Instant start = Instant.now();
+
+    for (List<Result> results : Iterables.partition(iterable, PARTITION_SIZE)) {
+      if (Thread.interrupted()) {
+        LOGGER.info("Catalog sync interrupted early, exiting");
+        session.getConsole().println("Catalog sync interrupted, exiting");
+        Thread.currentThread().interrupt();
+        throw new InterruptedException();
+      }
+
+      try {
+        solrClient.add(
+            results
+                .stream()
+                .map(Result::getMetacard)
+                .map(OfflineGazetteerPlugin::convert)
+                .collect(Collectors.toList()));
+      } catch (SolrServerException | IOException e) {
+        LOGGER.info("error while adding items to solr", e);
+        session.getConsole().printf("An error occured while syncing: %s", e.getMessage());
+        throw e;
+      }
+      counter += results.size();
```

Comment:
It would be nice yes 😄 , but it would require doing extra queries to find out the total number of results. Unless you have strong opinions on having the progress bar i'll leave this as a possible follow on/ later task.